src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -578,7 +721,14 @@ int RGWBucket::check_index_olh(rgw::sal::RadosStore* const rados_store,
     formatter->open_array_section("");
   }
 
-  const int max_shards = rgw::num_shards(bucket_info.layout.current_index);
+  const auto& index = bucket_info.layout.current_index;
+  if (index.layout.type != rgw::BucketIndexType::Normal) {
+    ldpp_dout(dpp, 0) << "ERROR: cannot check bucket indices with layouts of type " <<
+      current_layout_desc(bucket_info.layout) <<
+      " for bad OLH entries" << dendl;
+    return -EINVAL;
+  }
+  const int max_shards = rgw::num_shards(index);
```

Comment:
nit: since you've checked that the layout type is 'normal', you can call the non-asserting overload `rgw::num_shards(index.normal)`